src/main/java/com/ibm/narpc/NaRPCServerGroup.java

```diff
@@ -22,32 +22,50 @@
 package com.ibm.narpc;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.slf4j.Logger;
 
 public class NaRPCServerGroup<R extends NaRPCMessage, T extends NaRPCMessage> extends NaRPCGroup {
-	private NaRPCService<R,T> service;
-	private NaRPCDispatcher<R,T> dispatcher;
+	private static final Logger LOG = NaRPCUtils.getLogger();
+	public static final int DEFAULT_DISPATCHARRAY = 1;
+	
+	private ArrayList<NaRPCDispatcher<R,T>> dispatcherArray;
+	private AtomicInteger arrayIndex;
 	
 	public NaRPCServerGroup(NaRPCService<R,T> service) throws IOException{
-		super();
-		this.service = service;
-		this.dispatcher = new NaRPCDispatcher<R,T>(this, service);
-		Thread thread = new Thread(dispatcher);
-		thread.start();
-	}	
+		this(service, NaRPCGroup.DEFAULT_QUEUE_DEPTH, NaRPCGroup.DEFAULT_MESSAGE_SIZE, NaRPCGroup.DEFAULT_NODELAY, DEFAULT_DISPATCHARRAY);
+	}
 	
 	public NaRPCServerGroup(NaRPCService<R,T> service, int queueDepth, int messageSize, boolean nodelay) throws IOException{
+		this(service, queueDepth, messageSize, nodelay, DEFAULT_DISPATCHARRAY);
+	}	
+	
+	public NaRPCServerGroup(NaRPCService<R,T> service, int queueDepth, int messageSize, boolean nodelay, int arraySize) throws IOException{
 		super(queueDepth, messageSize, nodelay);
-		this.service = service;
-		this.dispatcher = new NaRPCDispatcher<R,T>(this, service);
-		Thread thread = new Thread(dispatcher);
-		thread.start();
+		this.dispatcherArray = new ArrayList<NaRPCDispatcher<R,T>>(arraySize);
+		for (int i = 0; i < arraySize; i++) {
+			NaRPCDispatcher<R,T> dispatcher = new NaRPCDispatcher<R,T>(this, service, i);
+			Thread thread = new Thread(dispatcher);
+			thread.start();
+			dispatcherArray.add(dispatcher);
+		}
+		this.arrayIndex = new AtomicInteger(0);
```

Comment:
Move AtomicInteger creation before thread start. This is a race condition when a thread already starts accepting connections before arrayIndex has been created it will result in a null pointer.

Reply:
My mistake, this cannot happen since no ServerEndpoint can be created while we are in the constructor of the Group.